tslearn/neighbors.py

```diff
@@ -109,8 +108,8 @@ class KNeighborsTimeSeries(KNeighborsTimeSeriesMixin, NearestNeighbors):
     >>> time_series = to_time_series_dataset([[1, 2, 3, 4], [3, 3, 2, 0], [1, 2, 2, 4]])
     >>> knn = KNeighborsTimeSeries(n_neighbors=1).fit(time_series)
     >>> dist, ind = knn.kneighbors(to_time_series_dataset([[1, 1, 2, 2, 2, 3, 4]]), return_distance=True)
-    >>> dist
-    array([[ 0.]])
+    >>> dist # doctest: +NORMALIZE_WHITESPACE
+    array([[0.]])
```

Comment:
Whitespace needed for doctest.

Reply:
I don't really understand this `# doctest: +NORMALIZE_WHITESPACE ` flag... Shouldn't it remove all whitespaces from both expected and received output? Seems like it still fails at times here locally, even with that flag...

Reply:
I think it fails because there are no whitespace.

I wonder if `+NORMALIZE_WHITESPACE` is useless in this situation, since the documentation of [NORMALIZE_WHITESPACE](https://docs.python.org/2/library/doctest.html#doctest.NORMALIZE_WHITESPACE) mentions that 
> When specified, all sequences of whitespace (blanks and newlines) are treated as equal.

I don't think that it does what you think it does (but I'm not very familiar with doctest, so I may be wrong :) )

Reply:
When I made the PR for the use of `numba` for DTW, I made some changes to the whitespaces because it failed locally, just to realize that these changes made the CI failed. The issue might have been a different version of doctest, but I didn't bother trying to understand where it came from.
You might need to wait for Romain's point of view if you need more details about that.

Reply:
The problem comes from the fact that numpy format for printing arrays has changed between version 1.13 and 1.14. On master branch, we use 1.13 version, but the plan is to switch to >=1.14 for tslearn 0.2 (cf dev branch). So I suggest that in this PR, you just make sure that tests pass (use the old formatting) and when we get to merging (which should be pretty soon now, I guess, you've already done a lot here!) I'll take care of that on my side*.

Romain

*: the thing is that maybe it would be better to merge that PR to dev, but I will do that since there will be many manual fixes to be done I guess

Reply:
And regarding `# doctest: +NORMALIZE_WHITESPACE` it is not sufficient since it does not ignore all whitespaces, it just considers 2 whitespaces being equivalent to a single one, but 0 whitespaces is not considered equivalent to 1 whitespace...